src/main/java/uk/gov/companieshouse/accounts/filing/service/file/validation/AccountsValidationServiceImpl.java

```
@@ -55,11 +55,20 @@ public Optional<AccountsValidatorStatusApi> validationStatusResult(final String
   @Override
   public void saveFileValidationResult(String accountFilingId, AccountsValidatorStatusApi accountStatus) {
     String fileId = accountStatus.fileId();
-    String accountType = accountStatus.resultApi().data().accountType();
-    requestFilingRepository.save(AccountsFilingRecord.validateResult(accountFilingId, fileId, accountType));
+    String accountsType = accountStatus.resultApi().data().accountType();
+    Optional<AccountsFilingEntry> entry = requestFilingRepository.findById(accountFilingId);
+    AccountsFilingEntry freshEntry;
+    if(entry.isPresent()){
+      freshEntry = entry.get();
+      freshEntry.setFileId(fileId);
+      freshEntry.setAccountsType(accountsType);
+    }else{
+      freshEntry = new AccountsFilingEntry(accountFilingId, fileId, accountsType, null, null, null);
```

Do we not want to throw an exception here, this endpoint shouldn't be called if there isn't already an entry